src/Processors/Transforms/BuildRuntimeFilterTransform.cpp

```diff
@@ -28,7 +28,10 @@ BuildRuntimeFilterTransform::BuildRuntimeFilterTransform(
 {
     const auto & filter_column = header_->getByPosition(filter_column_position);
     if (!filter_column_target_type->equals(*filter_column_original_type))
-        cast_to_target_type = createInternalCast(filter_column, filter_column_target_type, CastType::nonAccurate, {});
+    {
+        auto query_context = CurrentThread::get().getQueryContext();
+        cast_to_target_type = createInternalCast(filter_column, filter_column_target_type, CastType::nonAccurate, {}, query_context);
```

Comment:
If it's too complex to propagate real context here, let's use `nullptr`, better to avoid using `CurrentThread`.